src/parser/transform/statement/transform_create_table.cpp

```diff
@@ -95,6 +96,31 @@ unique_ptr<CreateStatement> Transformer::TransformCreateTable(duckdb_libpgquery:
 		case duckdb_libpgquery::T_PGColumnDef: {
 			auto cdef = (duckdb_libpgquery::PGColumnDef *)c->data.ptr_value;
 			auto centry = TransformColumnDefinition(cdef);
+			if (centry.type.IsSerial()) {
```

Comment:
This is the largest block and is the brains of the operation- happy to move it to a function and/or separate file if that is a better place for it to live

Reply:
That might be nicer indeed